### PR TITLE
Implement mongodb for order service

### PR DESCRIPTION
**Pull request by pascalbenschopTU**

### CODE REVIEW

**Comment by pascalbenschopTU** [comment]
The init-mongo.js script in /order is called the first time you call docker-compose. It inserts a user into the database from volume order_mongodb_data, if you want to redo the script you have to delete the volume in docker.

**Comment by pascalbenschopTU** [comment]
Sometimes the test failed with substract_stock 14 != 15, no clue why but rerunning the test fixes it.

File: order/app.py [hunk 1]
```diff
@@ -14,52 +13,47 @@
 
 app = Flask("order-service")
 
-db: redis.Redis = redis.Redis(host=os.environ['REDIS_HOST'],
-                              port=int(os.environ['REDIS_PORT']),
-                              password=os.environ['REDIS_PASSWORD'],
-                              db=int(os.environ['REDIS_DB']))
-
+app.config["MONGO_URI"] = 'mongodb://' + os.environ['MONGODB_USERNAME'] + ':' + os.environ['MONGODB_PASSWORD'] + '@' + os.environ['MONGODB_HOSTNAME'] + ':27017/' + os.environ['MONGODB_DATABASE']
```
Comment (Nobert1) [inline order/app.py:16]: Can i suggest using formatted string litterals instead? So f"mongodb://{os.environ['MONGODB_USERNAME']}:{os.environ['MONGODB_PASSWORD']}@{os.environ['MONGODB_HOSTNAME']}:27017/{os.environ['MONGODB_DATABASE']}". Doesn't make a difference i just think it's cleaner than using +

**Review by Nobert1 — CHANGES_REQUESTED** [review]

**Comment by LeonKempen** [comment]
> Sometimes the test failed with substract_stock 14 != 15, no clue why but rerunning the test fixes it.

Well that is something that we should look into. Otherwise this would mean that the db is sometimes not consistent with even one user. Do you know if this already occured before you made the mongodb changes?

**Comment by Nobert1** [comment]
> > Sometimes the test failed with substract_stock 14 != 15, no clue why but rerunning the test fixes it.
> 
> Well that is something that we should look into. Otherwise this would mean that the db is sometimes not consistent with even one user. Do you know if this already occured before you made the mongodb changes?

It did occur before the changes.
The current implementation only has "eventual" consistency. If the operation is performed, and the stock consumer has not yet processed the message to return the stock, and put it in the database before the next read operation, the stock will not be updated. So it's random if it passes this test or not at this given time. Adding a small sleep before reading the stock agian will probably fix this.

**Review by Nobert1 — APPROVED** [review]

**Pull request merged**